### PR TITLE
Turn owm_weather.js into a library that can be `require`d

### PR DESCRIPTION
**Pull request by cat-haines**
- Refactored JS so it's in a namespaced object
- Refactored property + method names in JS
- Decoupled AppMessage keys from library
- Allowed developers to set the first AppKey used
- Updated example app and docs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,29 +11,20 @@ Includes a simple test app as a proof of concept usage of a weather C API.
 
 * Obtain an API key from [OpenWeatherMap.org](http://home.openweathermap.org/users/sign_up).
 
-* Add these `AppMessage` keys to the `appKeys` array in `appinfo.json`:
+* Ensure `'enableMultiJS': true` is set in your appinfo.json.
+
+* Add `owm_weather/owm_weather.js` to `/src/js/lib`, include it in your your app.js file, and then instantiate an OWMWeather object.
```
Comment (C-D-Lewis) [inline README.md]: Is being in a `lib` directory required?

Comment (C-D-Lewis) [inline README.md]: s/app.js/`app.js`

File: README.md [hunk 2]
```diff
@@ -43,13 +34,30 @@ Pebble.addEventListener('appmessage', function(e) {
 #include "owm_weather/owm_weather.h"
 ```
 
-* Call `owm_weather_init()` to initialize the library when your app starts, supplying your API key.
+* Call `owm_weather_init(api_key)` to initialize the library when your app starts, supplying your API key.
 
 * Call `owm_weather_fetch()`, after PebbleKit JS is ready supplying a suitable
```
Comment (C-D-Lewis) [inline README.md]: Add a comma after 'ready'?

File: README.md [hunk 3]
```diff
@@ -43,13 +34,30 @@ Pebble.addEventListener('appmessage', function(e) {
...
   callback for events.
 
 That's it! When the fetch returns (successful or not), the callback will be called with a `OWMWeatherInfo` object for you to extract data from.
 
+### OWMWeather AppMessage Keys
```
Comment (C-D-Lewis) [inline README.md]: :100:

File: owm_weather/owm_weather.h [hunk 4]
```diff
@@ -48,6 +54,12 @@ typedef void(OWMWeatherCallback)(OWMWeatherInfo *info, OWMWeatherStatus status);
 //! @param api_key The API key or 'appid' from your OpenWeatherMap account.
 void owm_weather_init(char *api_key);
 
+//! Initialize the weather library with a base app_message key. The data is fetched after calling this, and should be accessed
```
Comment (C-D-Lewis) [inline owm_weather/owm_weather.h]: s/app_message/AppMessage

Comment (C-D-Lewis) [inline owm_weather/owm_weather.h]: Also wrap this line a bit tighter (80 or 100)

File: owm_weather/owm_weather.js [hunk 5]
```diff
@@ -1,61 +1,94 @@
...
-        'OWMWeatherAppMessageKeyBadKey': 1
-      });
+var OWMWeather = function(options) {
+  this._apiKey = '';
+
+  options = options || {};
+  this._appKeyBase = options.baseAppKey || 0;
+
+  this._appKeys = {
+    "Request": 0,
```
Comment (C-D-Lewis) [inline owm_weather/owm_weather.js]: Mixed " and ' used

File: test-app/src/js/app.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+var OWMWeather = require('lib/owm_weather.js');
+
+var owmWeather = new OWMWeather({ baseAppKey: 10 });
```
Comment (C-D-Lewis) [inline test-app/src/js/app.js]: The name of this variable differs from the example given in the readme. Do we care about that?

File: README.md (hunk 1 above)
Comment (cat-haines) [inline README.md]: Being in a lib folder isn't required, but we're also trying to guide users on best practices.

File: test-app/src/js/app.js (hunk 6 above)
Comment (cat-haines) [inline test-app/src/js/app.js]: good catch, i do care about that

**Comment by C-D-Lewis** [comment]
:+1: LGTM!

**Pull request merged**